ui-templates/custom-types/coordinates_xyz.json

```diff
@@ -0,0 +1,26 @@
+{
+    "coordinates_xyz": {
+        "python::module": "numpy",
+        "python::object": "array",
+        "parameters": {
+            "0": {
+                "name": "X",
+                "data_type": "number",
+                "description": "Provide the X dimension",
+                "required": true
```

Comment:
The use of required here is a little odd, because it's parent is not required. I was figuring I would use this required field to add the little red *, but it doesn't really make sense here because it makes it seem like the whole thing is required. Maybe we can figure out a different way to denote that if you enter one of the subfields you need them all. 
<img width="533" alt="Screen Shot 2022-06-29 at 10 37 07 AM" src="https://user-images.githubusercontent.com/5170636/176500800-6251edb9-2e9f-4502-bd55-57b5a477b7f0.png">